Desktop/html/js/analysis.js

```diff
@@ -632,15 +593,8 @@ JASPWidgets.AnalysisView = JASPWidgets.View.extend({
 
 	render: function () {
 
-		var results = this.model.get("results");
-
-		// once everything becomes jaspResults this is always an object and the following can be removed
-		var progress = this.model.get("progress")
-		if (typeof progress == "number")
-			this.model.set("progress", { value: progress, label: "" })
-		else if (!progress)
-			this.model.set("progress", { value: -1, label: "" })
-		// up to here
```

Comment:
I recall from a previous attempt to remove this, that `progress` is not always an object even now with jaspResults. So I'd keep this in here

Reply:
Or I can push a fix to jaspResults to always make it an object

Reply:
I think, if it isn't an object it will be `null` that should be fairly easy to check for.

Reply:
We at least have these to be mindful of it being null in:
https://github.com/jasp-stats/jasp-desktop/blob/0db025f5aa07b9a192599c60c407f1ec23fd7b74/Desktop/analysis/analysis.h#L80
and in the engine.cpp there seem to be a few `_progress					= -1;` hardcoded, although it could well be that the analysis render is never called in that case.

Reply:
Well, I didn't see those, but maybe I didnt look hard enough. I can just change those "-1" to null and then check for it in js.

Reply:
So I think the new commit should fix the progressbar blemishes.